src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx

```diff
@@ -30,31 +30,338 @@ const httpLink = new HttpLink({
         <Provider store={store}>
           <BrowserRouter>
             <I18nextProvider i18n={i18nForTest}>
-              {<AddOnStore />}
+              <AddOnStore />
             </I18nextProvider>
           </BrowserRouter>
         </Provider>
       </ApolloProvider>
     );
-    expect(getByTestId('AddOnEntryStore')).toBeInTheDocument();
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Example Plugin'));
+    });
+
+    waitFor(() => {
+      expect(screen.getByText('Example Plugin')).toBeInTheDocument();
+    });
+  });
+
+  test('renders AddOnEntry components with the correct information', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      expect(screen.getByText('Example Plugin')).toBeInTheDocument();
+      expect(screen.getByText('Created by: John Doe')).toBeInTheDocument();
+    });
+  });
+
+  test('searches for plugins and displays search results', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      userEvent.type(
+        screen.getByPlaceholderText('Search Name'),
+        'Sample Plugin'
+      );
+    });
+
+    waitFor(() => {
+      expect(
+        screen.getByText('Search results for Sample Plugin')
+      ).toBeInTheDocument();
+      expect(screen.getByText('Sample Plugin 1')).toBeInTheDocument();
+    });
   });
-  // const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
-  // const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
 
-  // beforeEach(() => {
-  //   useSelectorMock.mockClear();
-  //   useDispatchMock.mockClear();
-  // });
+  test('installs a plugin when "Install" button is clicked in AddOnEntry', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Install'));
+    });
+    waitFor(() => {
+      expect(screen.getByText('Install')).toBeInTheDocument();
```

Comment:
How does this imply "A plugin is installed?"

Reply:
It is working as it is expected, pls let me know if I am not getting your doubt.

Reply:
Same explanation as above.

Reply:
yes